WordPress/gutenberg
# Move linkUI to own custom appender in navigation block

## What?

When you click the Navigation Block appender, it automatically inserts a link. If you cancel the action (escape or close the modal without selecting a block), the automatically added link is removed and focus gets lost or yanked to the previous block. It's also jarring to click the appender, and then have a new thing inserted and also open a modal.

So, rather than insert something automatically and then replace it (especially in the add block flow), this uses the LinkUI as the appender. It saves a bit of code and the UX feels much cleaner. Also, there have been a lot of bugs due to the previous flow.

## Why?

Code quality, UX, and save ourselves the headaches of trying to manually manage when to insert and remove.

## How?

Create a new component, `NavigationLinkAppender` and use it to wrap the `<LinkUI/>` for inserting links. Then, use this on both the Navigation Block and Submenu blocks.

## Testing Instructions





### Testing Instructions for Keyboard


Especially test the keyboard flow. Add links and press Escape from the modal. You'll quickly see why this PR makes the keyboard implementation a lot better. Focus gets moved back to the appender instead of to the previous block.

## Screenshots or screencast 



**Before**

https://github.com/user-attachments/assets/25b7681f-bb2a-40bf-b502-4d5b135c69e6

**After**

https://github.com/user-attachments/assets/e5ed30fc-4e26-4094-b66d-1e5d3022f68e
